### PR TITLE
Add test code

### PR DESCRIPTION
**Pull request by aegilops**
Added `test` keys to every custom pattern.

1. allows testing efficiently in a testing script
2. provides a way to add test data to the `README.md` to cut/paste into the secret scanning UI (not done in this PR)

Every `patterns.yml` now has a passing `test` key, and some patterns have been amended to pass cleanly and have added `expected` data based on new test files.

### CODE REVIEW

**Comment by aegilops** [comment]
@GeekMasher - you'll need to update the schema please to allow the `test` key with key members `data` (required), and `start_offset`/`end_offset` (optional, integers, they default to 0 and -1 respectively)

File: README.md [hunk 1]
```diff
@@ -68,6 +68,177 @@ Custom Secret Scanning Patterns repository.
 - Arc
   
 
+### [IBANs](./pii/generate_iban)
```
Comment (GeekMasher) [inline README.md]: This is strange. These shouldn't be all published. This might be an issue with the tool, let me take a look.

Comment (aegilops) [reply README.md]: I didn't notice that. Thank you 🙏 

I recall there was a flag 🏴 to set those to be hidden, and I may have not included that in my branch, or the `markdown` mode doesn't do that as you intended.

Comment (aegilops) [reply README.md]: I've added `display: false` to the country-level IBANs `patterns.yml`, which has fixed this

**Review by GeekMasher — APPROVED** [review]
Amazing work!

**Pull request merged**